torchtext/experimental/datasets/raw/sequence_tagging.py

```diff
@@ -58,47 +59,43 @@ def _setup_datasets(dataset_name, separator, root=".data"):
         "valid": _construct_filepath(extracted_files, "dev.txt"),
         "test": _construct_filepath(extracted_files, "test.txt")
     }
-
-    datasets = []
-    for key in data_filenames.keys():
-        if data_filenames[key] is not None:
-            datasets.append(
-                RawTextIterableDataset(dataset_name, NUM_LINES[dataset_name],
-                                       _create_data_from_iob(data_filenames[key], separator)))
-        else:
-            datasets.append(None)
-
-    return datasets
+    return tuple(RawTextIterableDataset(dataset_name, NUM_LINES[dataset_name], _create_data_from_iob(data_filenames[item], separator)) if data_filenames[item] is not None else None for item in data_select)
 
 
-def UDPOS(*args, **kwargs):
+def UDPOS(root=".data", data_select=('train', 'valid', 'test')):
     """ Universal Dependencies English Web Treebank
 
     Separately returns the training and test dataset
 
     Arguments:
         root: Directory where the datasets are saved. Default: ".data"
+        data_select: a string or tuple for the returned datasets (Default: ('train', 'valid', 'test'))
+            By default, all the datasets (train, valid, test) are generated. Users could also choose any one or two of them,
+            for example ('train', 'valid', 'test') or just a string 'train'.
 
     Examples:
-        >>> from torchtext.datasets.raw import UDPOS
+        >>> from torchtext.experimental.datasets.raw import UDPOS
         >>> train_dataset, valid_dataset, test_dataset = UDPOS()
     """
-    return _setup_datasets(*(("UDPOS", "\t") + args), **kwargs)
+    return _setup_datasets("UDPOS", "\t", root=root, data_select=data_select)
 
 
-def CoNLL2000Chunking(*args, **kwargs):
+def CoNLL2000Chunking(root=".data", data_select=('train', 'test')):
     """ CoNLL 2000 Chunking Dataset
 
     Separately returns the training and test dataset
 
     Arguments:
         root: Directory where the datasets are saved. Default: ".data"
+        data_select: a string or tuple for the returned datasets (Default: ('train', 'test'))
+            By default, both datasets (train, test) are generated. Users could also choose any one or two of them,
+            for example ('train', 'test') or just a string 'train'.
 
     Examples:
-        >>> from torchtext.datasets.raw import CoNLL2000Chunking
-        >>> train_dataset, valid_dataset, test_dataset = CoNLL2000Chunking()
+        >>> from torchtext.experimental.datasets.raw import CoNLL2000Chunking
+        >>> train_dataset, test_dataset = CoNLL2000Chunking()
     """
-    return _setup_datasets(*(("CoNLL2000Chunking", " ") + args), **kwargs)
+    return _setup_datasets("CoNLL2000Chunking", " ", root=".data", data_select=('train', 'test'))
```

Comment:
data_select isn't forwarded here

Reply:
It's worrying that this wasn't caught by the tests (they all run green). Please expand coverage so that this error would have been caught.

Reply:
Will send out a PR and add tests for those scenarios.